src/main/java/org/opentripplanner/model/transfer/TransferPoint.java

```diff
@@ -1,80 +1,100 @@
  *   </li>
  *   <li>
- *     A {@link RouteTransferPoint} is a from/to point for a Route at the given stop. This only
- *     exist in GTFS, not in the Nordic NeTex profile. To support this we expand the route into
- *     all trips defined for it, and create {@link RouteTransferPoint} for each trip. We do the
- *     expansion because a Route may have more than on TripPattern and we want to use the stop
- *     position in pattern, not the stop for matching actual transfers. The reason is that
- *     real-time updates could invalidate a (route+stop)-transfer-point, since the stop could
- *     change to another platform(very common for railway stations). To account for this the
- *     RT-update would have to patch the (route&stop)-transfer-point. We simplify the RT-updates
- *     by converting expanding (route+stop) to (trip+stop position).
+ *     {@link StationTransferPoint} This apply to all trip stopping at a stop part of the given
+ *     station.
+ *     <p>The specificity-ranking is above {@link StationTransferPoint}s and less than
+ *     {@link RouteTransferPoint}.
+ *   </li>
+ *   <li>
+ *     A {@link RouteTransferPoint} is a from/to point for a Route at the given stop/station. This
+ *     only exist in GTFS, not in the Nordic NeTex profile.
  *
  *     <p>The specificity-ranking is above {@link StopTransferPoint}s and less than
  *     {@link TripTransferPoint}.
  *   </li>
  *   <li>
  *     {@link TripTransferPoint} A transfer from/to a Trip at the given stop position(not stop).
- *     GTFS Transfers specify a transfer from/to a trip and stop. But in OTP we map the stop to a
- *     stop position in pattern instead. This make sure that the transfer is still valid after a
- *     real-time update where the stop is changed. Especially for train stations changing the
- *     train platform is common and by using the stop position in pattern not the stop this
- *     become more robust. So, the OTP implementation follow the NeTEx Interchange definition
- *     here, not the GTFS specification.
  *
  *     <p>This is the most specific point type, and will override both {@link RouteTransferPoint}
  *     and {@link StopTransferPoint} if more than one match exist.
  *   </li>
  * </ol>
+ * <p>
+ * The GTFS Transfers may specify a transfer from/to a route/trip and stop/station. But in OTP we
+ * map the stop to a stop position in pattern. The OTP model {@link RouteTransferPoint} and
+ * {@link TripTransferPoint} do NOT reference the stop/station, but the
+ * {@code stopPositionInPattern} instead. The reason is that real-time updates could invalidate a
+ * (route+stop) transfer-point, since the stop could change to another  platform(common for railway
+ * stations). To account for this the RT-update would have to patch the (route&stop)-transfer-point.
+ * We simplify the RT-updates by converting expanding (route+stop) to (trip+stop position).
```

Comment:
What happens, if the RT pattern has shifted around the stop positions? Eg. Skipping a stop/station?

Reply:
Then it does not work. Similar problem arise if a Transfer is ref. a stop and the trip switches stop (e.g. change track in a railway station or platform in a bus terminal). Changing the transfer point to ref the station would solve this, but is not correct since we now allow the transfer to happen at all stops, not just the one specified. So, I think we should try this implementation out. Clean up the internal model, and make sure also transfers are updated by RT updates.